taichi/ir/ir_builder.h

```diff
@@ -185,6 +187,16 @@ class IRBuilder {
   BinaryOpStmt *create_cmp_eq(Stmt *l, Stmt *r);
   BinaryOpStmt *create_cmp_ne(Stmt *l, Stmt *r);
 
+  // Atomic operations.
+  AtomicOpStmt *create_atomic_add(Stmt *l, Stmt *r);
+  AtomicOpStmt *create_atomic_sub(Stmt *l, Stmt *r);
+  AtomicOpStmt *create_atomic_max(Stmt *l, Stmt *r);
+  AtomicOpStmt *create_atomic_min(Stmt *l, Stmt *r);
+  // Atomic bitwise operations.
+  AtomicOpStmt *create_atomic_and(Stmt *l, Stmt *r);
+  AtomicOpStmt *create_atomic_or(Stmt *l, Stmt *r);
+  AtomicOpStmt *create_atomic_xor(Stmt *l, Stmt *r);
```

Comment:
nit: Let's use `Stmt *dest, Stmt *val` instead of `Stmt *l, Stmt *r` in atomic operations.